cli/kaos_cli/services/tests/test_state_service.py

```diff
@@ -0,0 +1,228 @@
+
+    def test_state_service_list_providers(self):
+
+        # Act
+        provider_list = TestCase.state_service.list_providers()
+
+        # Assert
+        self.assertEqual([], provider_list)
+
+    @parameterized.expand(state_service_test_inputs)
+    def test_state_service_set_and_get(self, context, param):
+
+        # Arrange
+        if not isinstance(context, str):
+            TestCase.switch = True
+            if context is None:
+                context = {}
+            else:
+                context = str(context)
+
+        # Act
+        TestCase.state_service.set('section', context=context)
+
+        # Assert
+        self.assertEqual(context, TestCase.state_service.config['section']['context'])
+
+        # Act
+        TestCase.state_service.get('section', 'context')
+
+        # Assert
+        self.assertEqual(context, TestCase.state_service.config['section']['context'])
+
+    @parameterized.expand(state_service_test_inputs)
+    def test_state_service_set_and_get_sections(self, context, param):
+
+        # Arrange
+        if not isinstance(param, str):
+            TestCase.switch = True
+            if param is None:
+                param = {}
+            else:
+                param = str(context)
+
+        # Act
+        TestCase.state_service.set_section('section', 'subsection', param=param)
+
+        # Assert
+        self.assertEqual(param, TestCase.state_service.config['section']['subsection']['param'])
+
+        # Act
+        TestCase.state_service.get_section('section', 'subsection', 'param')
+
+        # Assert
+        self.assertEqual(TestCase.state_service.config['section']['subsection'],
+                         TestCase.state_service.has_section('section', 'subsection'))
+
+    def test_state_service_write(self):
+
+        # Act
+        TestCase.state_service.write()
+
+        # Assert
+        self.assertEqual(TestCase.state_service.config, ConfigObj(CONFIG_PATH))
+
+    @parameterized.expand(state_service_test_inputs)
+    def test_state_service_has_section(self, context, param):
+
+        # Arrange
+        TestCase.state_service.set('section', context=context)
+        TestCase.state_service.set_section('section', 'subsection', param=param)
+
+        if not TestCase.switch:
+            # Act
+            TestCase.state_service.has_section('section', 'subsection')
+
+            # Assert
+            self.assertEqual(TestCase.state_service.config['section']['subsection'],
+                             TestCase.state_service.has_section('section', 'subsection'))
+
+            # Act
+            TestCase.state_service.remove_section('section', 'subsection')
+
+            # Assert
+            self.assertEqual("removed",
+                             "removed" if 'subsection' not in TestCase.state_service.config['section'] else "not removed")
+
+    def test_state_service_remove(self):
+
+        # Act
+        TestCase.state_service.remove('section')
```

Comment:
This is an anti-pattern. Please create an instance of the `StateService` in each unit test and carry out the mutations you're testing for inside that scope of the test method only.

Reply:
The ```StateService``` now gets instantiated for each unit test as suggested.